Label filter not working

```
18:27:01.244 [http-nio-8080-exec-2] ERROR za.co.absa.commons.error.ErrorRef - ERROR_ID [b82616a2-b274-4937-992a-d53f9cdb4331] Query execution failed: 
================================================================================
WITH progress_view
FOR ee IN progress_view
    SEARCH ee._created <= @asAtTime
        AND (@timestampStart == null OR IN_RANGE(ee.timestamp, @timestampStart, @timestampEnd, true, true))
        AND (@applicationId == null OR @applicationId == ee.extra.appId)
        AND (@dataSourceUri == null OR @dataSourceUri == ee.execPlanDetails.dataSourceUri)
        AND (@writeAppends  == null OR ee.execPlanDetails.append IN @writeAppends)
          AND @lblValues[0] ANY == ee.labels['tags']
        AND (@searchTerm == null
            OR ANALYZER(LIKE(ee.execPlanDetails.frameworkName, CONCAT("%", TOKENS(@searchTerm, "norm_en")[0], "%")), "norm_en")
      OR ANALYZER(LIKE(ee.execPlanDetails.applicationName, CONCAT("%", TOKENS(@searchTerm, "norm_en")[0], "%")), "norm_en")
      OR ANALYZER(LIKE(ee.extra.appId, CONCAT("%", TOKENS(@searchTerm, "norm_en")[0], "%")), "norm_en")
      OR ANALYZER(LIKE(ee.execPlanDetails.dataSourceUri, CONCAT("%", TOKENS(@searchTerm, "norm_en")[0], "%")), "norm_en")
      OR ANALYZER(LIKE(ee.execPlanDetails.dataSourceType, CONCAT("%", TOKENS(@searchTerm, "norm_en")[0], "%")), "norm_en")
        )

    LET resItem = {
        "executionEventId" : ee._key,
        "executionPlanId"  : ee.execPlanDetails.executionPlanKey,
        "frameworkName"    : ee.execPlanDetails.frameworkName,
        "applicationName"  : ee.execPlanDetails.applicationName,
        "applicationId"    : ee.extra.appId,
        "timestamp"        : ee.timestamp,
        "dataSourceName"   : ee.execPlanDetails.dataSourceName,
        "dataSourceUri"    : ee.execPlanDetails.dataSourceUri,
        "dataSourceType"   : ee.execPlanDetails.dataSourceType,
        "append"           : ee.execPlanDetails.append,
        "durationNs"       : ee.durationNs,
        "error"            : ee.error
    }

    SORT resItem.@sortField @sortOrder
    LIMIT @pageOffset*@pageSize, @pageSize

    RETURN resItem
================================================================================
za.co.absa.spline.persistence.DatabaseException: Query execution failed: 
================================================================================
WITH progress_view
FOR ee IN progress_view
    SEARCH ee._created <= @asAtTime
        AND (@timestampStart == null OR IN_RANGE(ee.timestamp, @timestampStart, @timestampEnd, true, true))
        AND (@applicationId == null OR @applicationId == ee.extra.appId)
        AND (@dataSourceUri == null OR @dataSourceUri == ee.execPlanDetails.dataSourceUri)
        AND (@writeAppends  == null OR ee.execPlanDetails.append IN @writeAppends)
          AND @lblValues[0] ANY == ee.labels['tags']
        AND (@searchTerm == null
            OR ANALYZER(LIKE(ee.execPlanDetails.frameworkName, CONCAT("%", TOKENS(@searchTerm, "norm_en")[0], "%")), "norm_en")
      OR ANALYZER(LIKE(ee.execPlanDetails.applicationName, CONCAT("%", TOKENS(@searchTerm, "norm_en")[0], "%")), "norm_en")
      OR ANALYZER(LIKE(ee.extra.appId, CONCAT("%", TOKENS(@searchTerm, "norm_en")[0], "%")), "norm_en")
      OR ANALYZER(LIKE(ee.execPlanDetails.dataSourceUri, CONCAT("%", TOKENS(@searchTerm, "norm_en")[0], "%")), "norm_en")
      OR ANALYZER(LIKE(ee.execPlanDetails.dataSourceType, CONCAT("%", TOKENS(@searchTerm, "norm_en")[0], "%")), "norm_en")
        )

    LET resItem = {
        "executionEventId" : ee._key,
        "executionPlanId"  : ee.execPlanDetails.executionPlanKey,
        "frameworkName"    : ee.execPlanDetails.frameworkName,
        "applicationName"  : ee.execPlanDetails.applicationName,
        "applicationId"    : ee.extra.appId,
        "timestamp"        : ee.timestamp,
        "dataSourceName"   : ee.execPlanDetails.dataSourceName,
        "dataSourceUri"    : ee.execPlanDetails.dataSourceUri,
        "dataSourceType"   : ee.execPlanDetails.dataSourceType,
        "append"           : ee.execPlanDetails.append,
        "durationNs"       : ee.durationNs,
        "error"            : ee.error
    }

    SORT resItem.@sortField @sortOrder
    LIMIT @pageOffset*@pageSize, @pageSize

    RETURN resItem
================================================================================
	at za.co.absa.spline.persistence.ArangoImplicits$ArangoDatabaseAsyncScalaWrapper$$anonfun$queryAs$1.applyOrElse(ArangoImplicits.scala:91)
	at za.co.absa.spline.persistence.ArangoImplicits$ArangoDatabaseAsyncScalaWrapper$$anonfun$queryAs$1.applyOrElse(ArangoImplicits.scala:88)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
	at scala.util.Failure.recover(Try.scala:234)
	at scala.concurrent.Future.$anonfun$recover$1(Future.scala:395)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	at scala.concurrent.impl.CallbackRunnable.run$$$capture(Promise.scala:64)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
	at java.base/java.util.concurrent.ForkJoinTask.doExec$$$capture(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: com.arangodb.ArangoDBException: Response: 400, Error: 10 - AQL: failed to build filter while querying arangosearch view, query '{"type":"n-ary or","typeID":63,"subNodes":[{"type":"n-ary and","typeID":62,"subNodes":[{"type":"compare <=","typeID":28,"excludesNull":false,"subNodes":[{"type":"attribute access","typeID":35,"name":"_created","subNodes":[{"type":"reference","typeID":45,"name":"ee","id":0}]},{"type":"value","typeID":40,"value":1667496421180,"vType":"int","vTypeID":2}]},{"type":"value","typeID":40,"value":true,"vType":"bool","vTypeID":1},{"type":"value","typeID":40,"value":true,"vType":"bool","vTypeID":1},{"type":"value","typeID":40,"value":true,"vType":"bool","vTypeID":1},{"type":"value","typeID":40,"value":true,"vType":"bool","vTypeID":1},{"type":"value","typeID":40,"value":true,"vType":"bool","vTypeID":1},{"type":"array compare ==","typeID":65,"subNodes":[{"type":"indexed access","typeID":37,"subNodes":[{"type":"object","typeID":42,"subNodes":[{"type":"object element","typeID":43,"name":"array","subNodes":[{"type":"array","typeID":41,"sorted":false,"subNodes":[{"type":"array","typeID":41,"sorted":false,"subNodes":[{"type":"value","typeID":40,"value":"beer","vType":"string","vTypeID":4}]}]}]}]},{"type":"value","typeID":40,"value":0,"vType":"int","vTypeID":2}]},{"type":"attribute access","typeID":35,"name":"tags","subNodes":[{"type":"attribute access","typeID":35,"name":"labels","subNodes":[{"type":"reference","typeID":45,"name":"ee","id":0}]}]},{"type":"quantifier","typeID":73,"quantifier":"any"}]}]}]}': wrong value node type for Array comparison operator (while executing)
	at com.arangodb.internal.util.ResponseUtils.checkError(ResponseUtils.java:55)
	at com.arangodb.internal.velocystream.VstCommunication.checkError(VstCommunication.java:157)
	at com.arangodb.async.internal.velocystream.VstCommunicationAsync.lambda$execute$1(VstCommunicationAsync.java:80)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073)
	at com.arangodb.async.internal.velocystream.VstConnectionAsync.lambda$write$0(VstConnectionAsync.java:51)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
	at com.arangodb.internal.velocystream.internal.MessageStore.consume(MessageStore.java:62)
	at com.arangodb.internal.velocystream.internal.ChunkStore.checkCompleteness(ChunkStore.java:65)
	at com.arangodb.internal.velocystream.internal.ChunkStore.checkCompleteness(ChunkStore.java:59)
	at com.arangodb.internal.velocystream.internal.VstConnection.lambda$open$0(VstConnection.java:218)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

--- a/build/parent-pom/pom.xml
+++ b/build/parent-pom/pom.xml
@@ -384,7 +384,7 @@
             <dependency>
                 <groupId>za.co.absa.commons</groupId>
                 <artifactId>commons_${scala.compat.version}</artifactId>
-                <version>1.0.2</version>
+                <version>1.3.0</version>
             </dependency>
 
             <!-- Apache Commons -->

--- a/commons/src/main/scala/za/co/absa/spline/common/rest/RESTClientApacheHttpImpl.scala
+++ b/commons/src/main/scala/za/co/absa/spline/common/rest/RESTClientApacheHttpImpl.scala
@@ -37,7 +37,7 @@ class RESTClientApacheHttpImpl(
   (implicit ec: ExecutionContext)
   extends RESTClient {
 
-  import za.co.absa.commons.lang.OptionImplicits._
+  import za.co.absa.commons.lang.extensions.AnyExtension._
 
   override def get(path: String): Future[String] = execHttp {
     baseUri => new HttpGet(s"$baseUri/$path")

--- a/consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/controller/AbstractExecutionEventsController.scala
+++ b/consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/controller/AbstractExecutionEventsController.scala
@@ -25,7 +25,8 @@ class AbstractExecutionEventsController(
   repo: AbstractExecutionEventRepository
 ) {
 
-  import za.co.absa.commons.lang.OptionImplicits._
+  import za.co.absa.commons.lang.extensions.NonOptionExtension._
+  import za.co.absa.commons.lang.extensions.StringExtension._
 
   import scala.concurrent.ExecutionContext.Implicits._
 

--- a/consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/controller/LabelsController.scala
+++ b/consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/controller/LabelsController.scala
@@ -31,7 +31,7 @@ class LabelsController @Autowired()(
   val repo: LabelRepository
 ) {
 
-  import za.co.absa.commons.lang.OptionImplicits._
+  import za.co.absa.commons.lang.extensions.StringExtension._
 
   import scala.concurrent.ExecutionContext.Implicits._
 

--- a/consumer-services/src/main/scala/za/co/absa/spline/consumer/service/repo/DataSourceRepositoryImpl.scala
+++ b/consumer-services/src/main/scala/za/co/absa/spline/consumer/service/repo/DataSourceRepositoryImpl.scala
@@ -34,7 +34,7 @@ import scala.concurrent.{ExecutionContext, Future}
 @Repository
 class DataSourceRepositoryImpl @Autowired()(db: ArangoDatabaseAsync) extends DataSourceRepository {
 
-  import za.co.absa.commons.lang.OptionImplicits._
+  import za.co.absa.commons.lang.extensions.AnyExtension._
 
   override def getTimestampRange(
     asAtTime: Long,
@@ -89,10 +89,9 @@ class DataSourceRepositoryImpl @Autowired()(db: ArangoDatabaseAsync) extends Dat
         "includeNoWrite" -> Boolean.box(writeAppendOptions.contains(None)),
         "applicationId" -> maybeApplicationId.orNull,
         "dataSourceUri" -> maybeDataSourceUri.orNull
-      ).optionally(
-        _.updated("lblValues", _: Seq[Array[Label.Value]]),
-        lblValues.toSeq.asOption
-      )
+      ).when(lblValues.nonEmpty) {
+        _.updated("lblValues", lblValues)
+      }
     ).map { case Array(from, to) => from -> to }
   }
 
@@ -171,10 +170,9 @@ class DataSourceRepositoryImpl @Autowired()(db: ArangoDatabaseAsync) extends Dat
         "includeNoWrite" -> Boolean.box(writeAppendOptions.contains(None)),
         "applicationId" -> maybeWriteApplicationId.orNull,
         "dataSourceUri" -> maybeDataSourceUri.orNull
-      ).optionally(
-        _.updated("lblValues", _: Seq[Array[Label.Value]]),
-        lblValues.toSeq.asOption
-      ),
+      ).when(lblValues.nonEmpty) {
+        _.updated("lblValues", lblValues)
+      },
       new AqlQueryOptions().fullCount(true)
     ).map {
       arangoCursorAsync =>

--- a/consumer-services/src/main/scala/za/co/absa/spline/consumer/service/repo/ExecutionEventRepositoryImpl.scala
+++ b/consumer-services/src/main/scala/za/co/absa/spline/consumer/service/repo/ExecutionEventRepositoryImpl.scala
@@ -32,7 +32,7 @@ import scala.concurrent.{ExecutionContext, Future}
 @Repository
 class ExecutionEventRepositoryImpl @Autowired()(db: ArangoDatabaseAsync) extends ExecutionEventRepository {
 
-  import za.co.absa.commons.lang.OptionImplicits._
+  import za.co.absa.commons.lang.extensions.AnyExtension._
 
   override def getTimestampRange(
     asAtTime: Long,
@@ -84,10 +84,9 @@ class ExecutionEventRepositoryImpl @Autowired()(db: ArangoDatabaseAsync) extends
         "writeAppends" -> (if (writeAppendOptions.isEmpty) null else writeAppendOptions.flatten.map(Boolean.box)),
         "applicationId" -> maybeApplicationId.orNull,
         "dataSourceUri" -> maybeDataSourceUri.orNull
-      ).optionally(
-        _.updated("lblValues", _: Seq[Array[Label.Value]]),
-        lblValues.toSeq.asOption
-      ),
+      ).when(lblValues.nonEmpty) {
+        _.updated("lblValues", lblValues)
+      },
     ).map { case Array(from, to) => from -> to }
   }
 
@@ -163,10 +162,9 @@ class ExecutionEventRepositoryImpl @Autowired()(db: ArangoDatabaseAsync) extends
         "writeAppends" -> (if (writeAppendOptions.isEmpty) null else writeAppendOptions.flatten.map(Boolean.box)),
         "applicationId" -> maybeApplicationId.orNull,
         "dataSourceUri" -> maybeDataSourceUri.orNull
-      ).optionally(
-        _.updated("lblValues", _: Seq[Array[Label.Value]]),
-        lblValues.toSeq.asOption
-      ),
+      ).when(lblValues.nonEmpty) {
+        _.updated("lblValues", lblValues)
+      },
       new AqlQueryOptions().fullCount(true)
     ).map {
       arangoCursorAsync =>

--- a/persistence/src/main/scala/za/co/absa/spline/persistence/ArangoConnectionURL.scala
+++ b/persistence/src/main/scala/za/co/absa/spline/persistence/ArangoConnectionURL.scala
@@ -17,15 +17,16 @@
 package za.co.absa.spline.persistence
 
 import org.apache.commons.lang3.StringUtils.trimToNull
-import za.co.absa.commons.lang.OptionImplicits.StringWrapper
 import za.co.absa.spline.persistence.ArangoConnectionURL.ArangoSecureDbScheme
 
 import java.net.MalformedURLException
+import scala.collection.mutable
 import scala.util.matching.Regex
 
 case class ArangoConnectionURL(scheme: String, user: Option[String], password: Option[String], hosts: Seq[(String, Int)], dbName: String) {
 
-  import za.co.absa.commons.lang.OptionImplicits._
+  import za.co.absa.commons.lang.extensions.AnyExtension._
+  import za.co.absa.commons.lang.extensions.StringExtension._
 
   require(user.isDefined || password.isEmpty, "user cannot be blank if password is specified")
 
@@ -33,7 +34,7 @@ case class ArangoConnectionURL(scheme: String, user: Option[String], password: O
     val userInfo = trimToNull(Seq(user, password.map(_ => "*****")).flatten.mkString(":"))
     val commaSeparatedHostsString = hosts.map { case (host, port) => s"$host:$port" }.mkString(",")
 
-    new StringBuilder()
+    new mutable.StringBuilder()
       .append(s"$scheme://")
       .having(userInfo.nonBlankOption)(_ append _ append "@")
       .append(commaSeparatedHostsString)
@@ -45,6 +46,8 @@ case class ArangoConnectionURL(scheme: String, user: Option[String], password: O
 }
 
 object ArangoConnectionURL {
+
+  import za.co.absa.commons.lang.extensions.StringExtension._
 
   private[persistence] val ArangoDbScheme = "arangodb"
   private[persistence] val ArangoSecureDbScheme = "arangodbs"

--- a/persistence/src/main/scala/za/co/absa/spline/persistence/ArangoDatabaseFacade.scala
+++ b/persistence/src/main/scala/za/co/absa/spline/persistence/ArangoDatabaseFacade.scala
@@ -21,7 +21,6 @@ import com.arangodb.async.{ArangoDBAsync, ArangoDatabaseAsync}
 import com.arangodb.velocypack.module.scala.VPackScalaModule
 import org.slf4s.Logging
 import org.springframework.beans.factory.DisposableBean
-import za.co.absa.commons.lang.OptionImplicits.AnyWrapper
 import za.co.absa.commons.version.Version
 import za.co.absa.commons.version.impl.SemVer20Impl.SemanticVersion
 
@@ -31,6 +30,7 @@ import scala.concurrent._
 class ArangoDatabaseFacade(connectionURL: ArangoConnectionURL, maybeSSLContext: Option[SSLContext], activeFailover: Boolean)
   extends DisposableBean {
 
+  import za.co.absa.commons.lang.extensions.AnyExtension._
   import za.co.absa.spline.persistence.ArangoDatabaseFacade._
 
   private val ArangoConnectionURL(_, maybeUser, maybePassword, hostsWithPorts, dbName) = connectionURL

--- a/producer-model-mapper/src/main/scala/za/co/absa/spline/producer/modelmapper/v1_0/spark/SparkSplineExpressionConverter.scala
+++ b/producer-model-mapper/src/main/scala/za/co/absa/spline/producer/modelmapper/v1_0/spark/SparkSplineExpressionConverter.scala
@@ -26,7 +26,7 @@ class SparkSplineExpressionConverter(
   attrRefConverter: AttributeRefConverter
 ) extends ExpressionConverter {
 
-  import za.co.absa.commons.lang.OptionImplicits._
+  import za.co.absa.commons.lang.extensions.TraversableExtension._
 
   override def isExpression(obj: Any): Boolean = PartialFunction.cond(obj) {
     case exprDef: TypesV10.ExprDef@unchecked => exprDef

--- a/producer-services/src/main/scala/za/co/absa/spline/producer/service/model/ExecutionPlanPersistentModelBuilder.scala
+++ b/producer-services/src/main/scala/za/co/absa/spline/producer/service/model/ExecutionPlanPersistentModelBuilder.scala
@@ -20,8 +20,8 @@ import scalax.collection.Graph
 import scalax.collection.GraphEdge.DiEdge
 import scalax.collection.GraphPredef._
 import za.co.absa.commons.graph.GraphImplicits._
-import za.co.absa.commons.lang.CollectionImplicits._
-import za.co.absa.commons.lang.OptionImplicits._
+import za.co.absa.commons.lang.extensions.AnyExtension._
+import za.co.absa.commons.lang.extensions.TraversableOnceExtension._
 import za.co.absa.spline.persistence.DefaultJsonSerDe._
 import za.co.absa.spline.persistence.model.{Edge, EdgeDef}
 import za.co.absa.spline.persistence.{model => pm}

--- a/producer-services/src/test/scala/za/co/absa/spline/producer/service/model/ExecutionPlanPersistentModelBuilderSpec.scala
+++ b/producer-services/src/test/scala/za/co/absa/spline/producer/service/model/ExecutionPlanPersistentModelBuilderSpec.scala
@@ -24,7 +24,7 @@ import za.co.absa.spline.producer.service.model.ExecutionPlanPersistentModelBuil
 
 class ExecutionPlanPersistentModelBuilderSpec extends AnyFlatSpec with Matchers {
 
-  import za.co.absa.commons.lang.OptionImplicits._
+  import za.co.absa.commons.lang.extensions.TraversableExtension._
 
   behavior of "ExecutionPlanPersistentModelBuilder"
   behavior of "getSchemaInfos"

--- a/producer-services/src/test/scala/za/co/absa/spline/producer/service/repo/ExecutionPlanDeserFixAspectSpec.scala
+++ b/producer-services/src/test/scala/za/co/absa/spline/producer/service/repo/ExecutionPlanDeserFixAspectSpec.scala
@@ -72,7 +72,7 @@ class ExecutionPlanDeserFixAspectSpec
 
 object ExecutionPlanDeserFixAspectSpec {
 
-  import za.co.absa.commons.lang.OptionImplicits._
+  import za.co.absa.commons.lang.extensions.TraversableExtension._
 
   class ProceedingJoinPointSpy(args: AnyRef*) extends ProceedingJoinPoint {
     var capturedProceedingArgs: Seq[AnyRef] = _

